src/System.Text.Json/src/System/Text/Json/JsonDocument.cs

```diff
@@ -0,0 +1,542 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Text.Json
+{
+    public sealed partial class JsonDocument : IDisposable
+    {
+        private ReadOnlyMemory<byte> _utf8Json;
+        private CustomDb _parsedData;
+        private byte[] _extraRentedBytes;
+        private (int, string) _lastString = (-1, null);
```

Comment:
Change the name to indicate that it is a tuple (`_lastIndexAndString`?)

When reviewing  `(int lastIdx, string lastString) = _lastString;` (from https://github.com/dotnet/corefx/pull/34485/files#diff-d29f02d120fe0c5b1b74fb7da3b5b91bR197), I had to think twice and recall what `_lastString` actually was.